src/NServiceBus.AzureStoragePersistence.Tests/Timeouts/When_removing_timeouts_from_the_storage.cs

```diff
@@ -0,0 +1,319 @@
+            timeoutPersister.Add(timeout1);
+            timeoutPersister.Add(timeout2);
+
+            var timeouts = Test_Helper.Get_All_Timeouts_Using_GetNextChunk(timeoutPersister);
+            Assert.IsTrue(timeouts.Count == 2);
+
+            var itemRemoved = true;
+            foreach (var timeout in timeouts)
+            {
+                itemRemoved &= timeoutPersister.TryRemove(timeout.Item1);
+            }
+
+            Assert.IsTrue(itemRemoved, "Expected 2 ivocations to return true, but one or both of them returned false");
+
+            Test_Helper.Assert_All_Timeouts_Have_Been_Removed(timeoutPersister);
+
+            Test_Helper.Cleanup_Storage_Account(timeout1, timeoutPersister, false);
+        }
+
+        [Test]
+        public void Should_return_false_if_timeout_already_deleted_for_TryRemove_invocation()
+        {
+            var timeoutPersister = Test_Helper.Create_TimeoutPersister();
+
+            var timeout = Test_Helper.Generate_Timeout_With_Headers();
+            timeoutPersister.Add(timeout);
+
+            var timeouts = Test_Helper.Get_All_Timeouts_Using_GetNextChunk(timeoutPersister);
+            Assert.IsTrue(timeouts.Count == 1);
+
+            var timeoutId = timeouts.First().Item1;
+
+            Assert.IsTrue(timeoutPersister.TryRemove(timeoutId));
+            Assert.IsFalse(timeoutPersister.TryRemove(timeoutId));
+
+            Test_Helper.Cleanup_Storage_Account(timeout, timeoutPersister, false);
+        }
+
+        [Test]
+        public void Should_remove_timeouts_by_sagaid()
+        {
+            var timeoutPersister = Test_Helper.Create_TimeoutPersister();
+            var sagaId1 = Guid.NewGuid();
+            var sagaId2 = Guid.NewGuid();
+            var timeout1 = Test_Helper.Getnerate_Timeout_With_Saga_Id(sagaId1);
+            var timeout2 = Test_Helper.Getnerate_Timeout_With_Saga_Id(sagaId2);
+            timeoutPersister.Add(timeout1);
+            timeoutPersister.Add(timeout2);
+
+            var timeouts = Test_Helper.Get_All_Timeouts_Using_GetNextChunk(timeoutPersister);
+            Assert.IsTrue(timeouts.Count == 2);
+
+            timeoutPersister.RemoveTimeoutBy(sagaId1);
+            timeoutPersister.RemoveTimeoutBy(sagaId2);
+
+            Test_Helper.Assert_All_Timeouts_Have_Been_Removed(timeoutPersister);
+
+            Test_Helper.Cleanup_Storage_Account(timeout1, timeoutPersister, false);
+        }
+
+        [Test]
+        public async Task TryRemove_should_work_with_concurrent_operations()
+        {
+            var timeoutPersister = Test_Helper.Create_TimeoutPersister();
+            var timeout = Test_Helper.Generate_Timeout_With_Headers();
+            timeoutPersister.Add(timeout);
+
+            var task1 = Task.Factory.StartNew(() => timeoutPersister.TryRemove(timeout.Id));
+            var task2 = Task.Factory.StartNew(() => timeoutPersister.TryRemove(timeout.Id));
+
+            await Task.WhenAll(task1, task2).ConfigureAwait(false);
```

Comment:
@danielmarbach is this going to be a true concurrent execution of tasks or could be done better?